server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -1049,6 +1054,11 @@ public void cancel() {
                 "cancel must be called after the requestId [" + requestId + "] has been removed from clientHandlers";
             FutureUtils.cancel(future);
         }
+
+        @Override
+        public String toString() {
+            return "TimeoutHandler for [" + requestId + "][" + action + "]";
```

Comment:
One nit: `TimeoutHandler` -> `timeout handler` so that we're speaking plain English instead of code in these messages

Reply:
I am okay with you pushing this directly.